### PR TITLE
Including unit test

### PR DESCRIPTION
**Pull request by supritat**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### CODE REVIEW

File: core/jazz_slack-channel/test/test.js [hunk 1]
```diff
@@ -1,11 +1,1025 @@
...
+const sinon = require('sinon');
+const request = require('request');
+const format = require("string-template");
+
 const index = require('../index');
+const configObj = require('../components/config.js');
+
+describe('jazz_slack-channel', function () {
+  var err, event, context, callback, callbackObj, reqStub;
+  beforeEach(function () {
```
Comment (svidhani) [inline core/jazz_slack-channel/test/test.js]: new line here

File: core/jazz_slack-channel/test/test.js [hunk 2]
```diff
@@ -1,11 +1,1025 @@
...
+      });
+    });
+
+    it("should indicate input error if service event.method is not defined/invalid", () => {
+      let invalidArray = ["", null, undefined];
+      invalidArray.forEach(each => {
+        event.method = each;
+        index.genericInputValidation(event, config)
+        .catch(error => {
+          expect(error).to.include({ result: 'inputError', message: 'Service inputs not defined!' })
```
Comment (svidhani) [inline core/jazz_slack-channel/test/test.js]: ideally we should provide a better error message which input is not defined. Service inputs not defined - 'method'
This is more of a change in the source code. What do you think?

File: core/jazz_slack-channel/test/test.js [hunk 3]
```diff
@@ -1,11 +1,1025 @@
...
+
+    it("should indicate input validation error/ BadRequest", () => {
+      let inputError = {
+        result: "inputError",
+        message: "Provided invalid input"
+      };
+      let genericInputValidation = sinon.stub(index, 'genericInputValidation').rejects(inputError);
+
+      index.handler(event, context, (err, res) => {
+        expect(err).to.include('{"errorType":"BadRequest","message":"'+inputError.message+'"}');
```
Comment (svidhani) [inline core/jazz_slack-channel/test/test.js]: give spaces ("' + '" )
fix in other places as well

File: core/jazz_slack-channel/test/test.js (hunk 1 above)
Comment (supritat) [reply core/jazz_slack-channel/test/test.js]: Included

File: core/jazz_slack-channel/test/test.js (hunk 2 above)
Comment (supritat) [reply core/jazz_slack-channel/test/test.js]: Changes made in source code.

File: core/jazz_slack-channel/test/test.js (hunk 3 above)
Comment (supritat) [reply core/jazz_slack-channel/test/test.js]: Done

File: core/jazz_slack-channel/index.js [hunk 4]
```diff
@@ -27,17 +27,17 @@ const logger = require("./components/logger.js")();
 const request = require("request");
 const format = require("string-template");
```
[the remark this answers is not in the document]
Comment (supritat) [reply core/jazz_slack-channel/index.js]: Since string value is driven from config file, won't be able to make use of JS template literals.

**Pull request merged**